src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { MpcWalletContext } from '@/components/common/ConnectWallet/MPCWalletProvider'
+import CopyButton from '@/components/common/CopyButton'
+import ModalDialog from '@/components/common/ModalDialog'
+import { Box, Button, DialogContent, DialogTitle, IconButton, TextField, Typography } from '@mui/material'
+import { useContext, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { Visibility, VisibilityOff, Close } from '@mui/icons-material'
+import css from './styles.module.css'
+import ErrorCodes from '@/services/exceptions/ErrorCodes'
+import { logError } from '@/services/exceptions'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+
+enum ExportFieldNames {
+  password = 'password',
+  pk = 'pk',
+}
+
+type ExportFormData = {
+  [ExportFieldNames.password]: string
+  [ExportFieldNames.pk]: string | undefined
+}
+
+const ExportMPCAccountModal = ({ onClose, open }: { onClose: () => void; open: boolean }) => {
+  const { exportPk } = useContext(MpcWalletContext)
+  const [error, setError] = useState<string>()
```

Comment:
nit: Just found out we can even use RHF to display an error separate from its input https://react-hook-form.com/docs/useformstate/errormessage

Reply:
Although on second thought if we set the error via RHF it will not be persisted afair so this is probably better.